app/user.go

```diff
@@ -1251,11 +1251,15 @@ func (a *App) UpdateHashedPassword(user *model.User, newHashedPassword string) *
 }
 
 func (a *App) ResetPasswordFromToken(userSuppliedTokenString, newPassword string) *model.AppError {
+	return a.resetPasswordFromToken(userSuppliedTokenString, newPassword, model.GetMillis())
```

Comment:
A word on this, because `go` does not allow monkey-patching, the easiest way* to "mock" the time is to have it as an argument. SO the "Public" method is unchanged but having the code in a private method allows us to mock the time.

* the hard way/"don't touch my production code way" would be to have the `App` struct hold a `Clock` interface that depends on `time` by default but would be mockable... which is quite a huge refactor 😅

Reply:
> Sometimes, the elegant implementation is just a function. Not a method. Not a class. Not a framework. Just a function.

-- John Carmack